### PR TITLE
metall: initial recipe

### PR DESCRIPTION
**Pull request by bigerl**
Specify library name and version:  [metall/0.20](https://github.com/LLNL/metall)

At [DICE Group](https://github.com/dice-group/) we need it as a dependency for some of our software assets.

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/metall/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+        "use_anonymous_new_map": [True, False],
+        # Experimental: Build programs that require the NUMA policy library (numa.h)
+        "build_numa": [True, False],
+        # Experimental: Try to free the associated pages and file space when objects equal to or larger than that
+        # is deallocated
+        "free_small_object_size_hint": "ANY",
+    }
+
+    default_options = {
+        "boost:header_only": True,
```
Comment (uilianries) [inline recipes/metall/all/conanfile.py:41]: Is boost header only mandatory?

File: recipes/metall/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+        "disable_small_object_cache": False,
+        "use_sorted_bin": False,
+        "default_vm_reserve_size": 0,
+        "max_segment_size": 0,
+        "initial_segment_size": 0,
+        "use_anonymous_new_map": False,
+        "build_numa": False,
+        "free_small_object_size_hint": 0,
+    }
+    requires = "boost/1.75.0",
```
Comment (uilianries) [inline recipes/metall/all/conanfile.py:52]:
```suggestion
    requires = "boost/1.79.0",
```
Try to use the latest version available if it does not break

File: recipes/metall/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+    def _compilers_minimum_version(self):
+        return {
+            "gcc": "8.3",
+            "clang": "9",
+        }
+
+    def validate(self):
+        if self.settings.compiler.get_safe("cppstd"):
+            tools.check_min_cppstd(self, self._min_cppstd)
+        if self.settings.os == "Windows":
```
Comment (uilianries) [inline recipes/metall/all/conanfile.py:73]:
```suggestion
        if self.settings.os != "Linux":
```
According the message error, requires Linux

**Review by uilianries — COMMENTED** [review]
Thank you for your contribution

File: recipes/metall/all/conanfile.py (hunk 1 above)
Comment (bigerl) [reply recipes/metall/all/conanfile.py:41]: No. It's just the minimum subset of boost required. I conclude, it should be removed then?

Comment (uilianries) [reply recipes/metall/all/conanfile.py:41]: Yes, keep the default options always, so remove it. Forcing different options, result in different behavior, not expected by users. Also, you may find package options which are not built/available in Conan center, which makes your package broken.

File: recipes/metall/all/conanfile.py (hunk 3 above)
Comment (bigerl) [reply recipes/metall/all/conanfile.py]: I've double checked the code. It has preprocessor switches for Linux and Macos and requires some POSIX functionalities (especially mmap). I would say, allowing only Linux and Macos is reasonable.

File: recipes/metall/all/conanfile.py (hunk 1 above)
Comment (bigerl) [reply recipes/metall/all/conanfile.py]: makes sense. I removed it.

File: recipes/metall/all/test_package/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    def layout(self):
+        cmake_layout(self)
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+
+    def test(self):
+        self.run(os.path.join(self.cpp.build.bindirs[0], "test_package"), run_environment=True)
```
Comment (SSE4) [inline recipes/metall/all/test_package/conanfile.py:26]:
```suggestion
        if not tools.cross_building(self):
            self.run(os.path.join(self.cpp.build.bindirs[0], "test_package"), run_environment=True)
```

**Review by jgsogo — APPROVED** [review]

**Pull request merged**